### PR TITLE
🚧Setup CICD for project

### LINKED ISSUE #1: 🚧Setup CICD for project
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Update the _**CICD**_ dotnet tool to the latest version.   This will require various _**dotnet**_ commands to be able to set up the _**kinsondigital.cicd**_ dotnet tool that is globally in NuGet.

💡What is a **_dotnet_** tool?
A dotnet tool is nothing but a console application that anybody can create that you can easily install and use in another dotnet application.  As long as the _**dotnet**_ tool is publicly available in nuget.org, it can be automatically pulled into your computer or installed globally on your machine to use!!  It is free and make it easy to distribute tools for anything from simple one off project use cases to complicated CICD automation.

### **Installing CICD**
1. Create a manifest file.
    > **Note** This is nothing but a configuration file. This config file is just a list of all the **_dotnet_** tools that are setup in a project for the _**dotnet**_ ecosystem to use.
    1. If you do not have the **_dotnet_** CLI installed on your machine, we first need to get that installed.  First check to see if you have the CLI installed by opening the windows terminal in ANY location.  Execute the command `dotnet --version`.  If you see a version number then you have it installed.
    2. Next, create a tool manifest file.  This will create an barebones file in root of the project so you can installed _**dotnet**_ tools.  
        To do this, run the command `dotnet new tool-manifest` the root of the repository.  Once it is complete, you will see a _**.config**_ folder created in the root of the repo for you and inside you should see a **_dotnet-tools.json_** file inside it.
2. Now it is time to add a dotnet tool to the project.  
    1. To create a dotnet tool, use the command `dotnet tool install kinsondigital.cicd --version 1.0.0-preview.20`.
        This should update the _**dotnet-tools.json**_ manifest file with additional information.  That info is the name of the tool as well as the version.  This gives the user the ability to run the `dotnet tool restore` command in the same location as the _**.config**_ folder at any time and it will automatically install the tool for that project so it is ready to use.

### **CICD Configuration**

This configuration is what **_CICD_** uses to setup some of its behavior when it runs.  Follow the steps below to set this up.

1. Create a folder in the root of the repository with the name _**.nuke**_.
2. Inside of the _**.nuke**_ folder, create a new file with the name _**parameters.json**_.
3. Add the JSON content below to the file from step 2.
```json
{
    "$schema": "./build.schema.json",
    "Solution": "Partithyst.sln",
    "RepoOwner": "KinsonDigital",
    "RepoName": "Partithyst",
    "ProjectName": "Partithyst",
    "PreProcessReadMe": true
}
```
4. If everything is setup correctly, you should be able to run the `dotnet cicd --help` and see results similar to the image below.
Image
5. Auto generate the workflow files using a CICD command.
    1. This can be done by executing the `dotnet cicd GenerateWorkflows --workflow-template-output "<FULL FILE PATH>"`
        > **Note** The path to generate will be a directory path and it MUST BE A FULLY QUALIFIED PATH.  Use the full path to the **_.github/workflows_** directory.  If this does not exist, create it first then run the command.
7. To create the badges, just copy prev and prod PR status badges from Velaptor to this project.
> **Note** Once you copy the badges over, just change the name of the projects in the URLs.  Make sure that the names of the workflow _***.yml**_ files in the URL are correct.  These are the workflows you generated in step 5.

### Acceptance Criteria

- [x] CICD setup for the project
- [x] Badges for the prev and prod release PR status check setup in the README file.
- [x] Workflows generated using the CICD dotnet tool.

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [X] Issue linked to the correct project _(if applicable)_.
- [X] Issue linked to the correct milestone _(if applicable)_.
- [x] Draft pull request created and linked to this issue _(only required with code changes)_.

### Issue Dependencies

_No response_

### Related Work

_No response_

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| CICD Changes        | `♻️cicd` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### PR DESCRIPTION
**Pull request by kselena**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/feature/<issue-num>-<description>
      Example: preview/feature/123-my-preview-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_preview release_** branch, with the correct naming syntax, in the following example:
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
    Example: preview/v1.2.3-preview.4
    ```
  </details>

- [x] The **_base(target)_** branch for this pull request is a **_preview release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

💡For more information on branching, refer to the project's [branching documentation](https://github.com/KinsonDigital/.github/blob/master/docs/Branching.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title matches the title of the associated issue.

- [x] A `preview` label has been added to the pull request.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out, and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

File: .config/dotnet-tools.json
Comment (KinsonDigitalAdmin) [inline .config/dotnet-tools.json:1]: Add a single empty line to the bottom of this file.

File: .github/workflows/feature-pr-status-check.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+on:
+  pull_request:
+    branches:
+      - develop
+
+
+jobs:
+  feature_pr_status_check:
+    if: startsWith(github.head_ref, 'dependabot/nuget/') == false # Do not run with dependabot branches
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/feature-pr-status-check.yml]: Remove this line.  This is residual from the dependabot stuff.  But since we are removing dependabot, it needs to be removed.

```suggestion
```

File: .github/workflows/hotfix-pr-status-check.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+on:
+  pull_request:
+    branches:
+      - master
+
+
+jobs:
+  hot_fix_pr_status_check:
+    if: contains(github.ref, 'refs/pull') && startsWith(github.head_ref, 'hotfix/')
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/hotfix-pr-status-check.yml]: Remove this line.  This is residual from the dependabot stuff.  But since we are removing dependabot, it needs to be removed.

```suggestion
```

File: .github/workflows/prev-feature-pr-status-check.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+on:
+  pull_request:
+    branches:
+      - preview/v*.*.*-preview.*
+
+
+jobs:
+  preview_feature_pr_status_check:
+    if: startsWith(github.head_ref, 'dependabot/nuget/') == false # Do not run with dependabot branches
```
Comment (KinsonDigitalAdmin) [inline .github/workflows/prev-feature-pr-status-check.yml]: Remove this line.  This is residual from the dependabot stuff.  But since we are removing dependabot, it needs to be removed.

```suggestion
```

File: .nuke/parameters.json
Comment (KinsonDigitalAdmin) [inline .nuke/parameters.json:1]: Add a single empty line to this file

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

File: .config/dotnet-tools.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "version": 1,
+  "isRoot": true,
+  "tools": {
+    "kinsondigital.cicd": {
+      "version": "1.0.0-preview.20",
```
Comment (KinsonDigitalAdmin) [inline .config/dotnet-tools.json]: Update this to the latest version.
There have been some releases since you worked on this issue.

```suggestion
      "version": "1.0.0-preview.22",
```

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]
Remove the following workflows.  
I have no idea how these made it in.  :)

image

**Review by KinsonDigitalAdmin — APPROVED** [review]
